sources/api/thar-be-settings/src/service.rs

```diff
@@ -135,8 +155,21 @@ impl ServiceRestart for model::Service {
             trace!("Args: {:?}", &command_strings);
 
             // Go execute the restart command
-            let result = process::Command::new(command)
-                .args(command_strings)
+            let mut process_command = process::Command::new(command);
+            process_command.args(command_strings);
+            match changed_settings {
+                Some(value) => {
+                    let changed_settings_string = value.iter().join(" ");
+                    info!(
+                        "restarting service  with changed settings {:?}",
+                        changed_settings_string
+                    );
+                    process_command.env("changed_settings", changed_settings_string);
+                }
+                None => {}
+            }
```

Comment:
Can you try?

```rust
let process_command = if let Some(changed_settings_string) = changed_settings_string {
    process::Command::new(command)
         .args(command_strings)
         .env("changed_settings", changed_settings_string.iter().join(" "));
} else {
    process::Command::new(command).args(command_strings)
}
```

Reply:
Yeah sure. I had moved the common part to the top, but with this approach, the empty `None` would go away. I will try it. Thanks!